solidity/test/KeepFactorySelectionTest.js

```diff
@@ -212,27 +360,193 @@ describe("KeepFactorySelection", async () => {
+    it("completes when KEEP stake factory address is zero", async () => {
+      await keepStakedVendor.setFactory(constants.ZERO_ADDRESS)
+
+      await keepFactorySelection.setMinimumBondableValue(newValue, 5, 3)
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(
+        defaultValue,
+      )
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(
+        defaultValue,
+      )
+    })
+
+    // KEEP stake vendor set, factory set.
+    it("updates value in KEEP stake factory", async () => {
+      await keepFactorySelection.setMinimumBondableValue(newValue, 5, 3)
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(newValue)
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(
+        defaultValue,
+      )
+    })
+
+    // ETH-only vendor set, factory not set.
+    it("updates value in ETH-only factory", async () => {
+      await keepFactorySelection.setFullyBackedKeepVendor(
+        fullyBackedVendor.address,
+      )
+      await fullyBackedVendor.setFactory(constants.ZERO_ADDRESS)
+
+      await keepFactorySelection.setMinimumBondableValue(newValue, 5, 3)
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(newValue)
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(
+        defaultValue,
+      )
+    })
+
+    // ETH-only vendor set, factory set.
+    it("updates value in ETH-only factory", async () => {
+      await keepFactorySelection.setFullyBackedKeepVendor(
+        fullyBackedVendor.address,
+      )
+
+      await keepFactorySelection.setMinimumBondableValue(newValue, 5, 3)
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(newValue)
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(newValue)
+    })
+
+    // KEEP stake vendor set, factory set.
+    // ETH-only vendor set, factory not set.
+    it("updates value in locked factories", async () => {
+      await keepFactorySelection.setFullyBackedKeepVendor(
+        fullyBackedVendor.address,
+      )
+
+      // Lock factories
+      await keepFactorySelection.lockFactoriesVersions(
+        keepStakedFactory.address,
+        fullyBackedFactory.address,
+      )
+
+      // Upgrade factories in vendors
+      const newKeepFactory = await ECDSAKeepFactoryStub.new()
+      await keepStakedVendor.setFactory(newKeepFactory.address)
+
+      const newEthFactory = await ECDSAKeepFactoryStub.new()
+      await fullyBackedVendor.setFactory(newEthFactory.address)
+
+      // Set minimum bondable value
+      await keepFactorySelection.setMinimumBondableValue(newValue, 5, 3)
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(newValue)
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(newValue)
+
+      expect(await newKeepFactory.minimumBondableValue()).to.eq.BN(defaultValue)
+      expect(await newEthFactory.minimumBondableValue()).to.eq.BN(defaultValue)
+    })
```

Comment:
I think we are missing a test case when:
- KEEP stake vendor is set, factory is set
- ETH-only vendor is set, factory is set
- versions of factories are locked

Reply:
> I think we are missing a test case when:

There are test cases covering these scenarios:

>     * KEEP stake vendor is set, factory is set
`updates value in KEEP stake factory`

>     * ETH-only vendor is set, factory is set
`updates value in ETH-only factory`

>     * versions of factories are locked
`updates value in locked factories`